src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets

```diff
@@ -28,4 +30,42 @@ This file contains targets specific for Android application projects.
 
   <Target Name="Run" DependsOnTargets="$(_RunDependsOn)" />
 
+  <PropertyGroup>
+    <_RunWithLoggingDependsOn>
+      Install;
+    </_RunWithLoggingDependsOn>
+
+    <_RunLogFilePath Condition=" '$(_RunLogFilePath)' == '' ">$(IntermediateOutputPath)logcat.txt</_RunLogFilePath>
+    <_RunLogVerbose Condition=" '$(_RunLogVerbose)' == '' ">false</_RunLogVerbose>
+  </PropertyGroup>
+
+  <Target Name="RunWithLogging"
+          DependsOnTargets="$(_RunWithLoggingDependsOn)">
+    <GetAndroidActivityName
+        Condition=" '$(AndroidLaunchActivity)' == '' "
+        ManifestFile="$(IntermediateOutputPath)android\AndroidManifest.xml">
+      <Output TaskParameter="ActivityName" PropertyName="AndroidLaunchActivity" />
+    </GetAndroidActivityName>
+
+    <PropertyGroup>
+      <_MonoLog>default,assembly,timing=bare</_MonoLog>
+      <_MonoLog Condition=" '$(_RunLogVerbose)' != 'false' ">$(_MonoLog),mono_log_level=debug,mono_log_mask=all</_MonoLog>
+      <_SleepTime>1000</_SleepTime> <!-- milliseconds -->
```

Comment:
Should we make this a public-facing MSBuild property? So, customers could pass `-p:Duration=10000`? I could see that an app might take a few seconds before it crashes, and this would be helpful to grab the log.

Reply:
Good idea, however I think `Duration` is too generic a name. Will think of something.